### ISSUE TITLE
DEV: Document and rename AIPlayers

### ISSUE THREAD

**Issue by Alex-Vasile** [magefree/mage#9179]
The current state of the AI players is not great. The classes are fairly impenetrable, down the the names `ComputerPlayer2`, `ComputerPlayer7`.

They need a lot more documentation, and especially a description of how each works and the differences between them.

**Comment by JayDi85** [comment]
That's diff versions must be related to improves and additional features. Each AI version adds more features. So `ComputerPlayer7` is only one real AI player with all possible features. If something uses old classes then it must be researched -- maybe it can be upgraded to improved AI class or simplified.

**Comment by JayDi85** [comment]
Yes, it can be merged together in one class. But after research. As
example: I remember some ai choices or auto-commands used simplified
computer player class (maybe v2).

Also new code must be compatible with new AI versions, e. g. with potential
machine learning features (ML).

Hmm. Maybe it's a good thing to split technical work from decision related
features (current classes), so it's possible to add improved features
without big changes in other logic (player class implementation is hard and
needs many features/methods for workable code).
